pkg/summary/options.go

```diff
@@ -0,0 +1,51 @@
+package summary
+
+import (
+	"github.com/fairwindsops/goldilocks/pkg/kube"
+	"github.com/fairwindsops/goldilocks/pkg/utils"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Functional options
+type Option func(*Options)
```

Comment:
Never seen this pattern before. Very cool

Reply:
I found it a while back and have fallen in love with it. It's just so useful for all sorts of packages and `NewFoo()` functions 😄 Here's a blog post about it if you're interested: https://www.sohamkamani.com/golang/options-pattern/

Reply:
Agreed, I like this a lot!